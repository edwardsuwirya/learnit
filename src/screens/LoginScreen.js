import React from 'react';
import { StyleSheet, SafeAreaView, View, Text, TextInput, TouchableOpacity } from 'react-native';

const LoginScreen = () => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.body}>

        <Text style={styles.title}>User Name</Text>
        <TextInput style={styles.input}></TextInput>
        <Text style={styles.title}>Password</Text>
        <TextInput style={styles.input} secureTextEntry={true}></TextInput>
        <TouchableOpacity
          style={styles.button}
        >
          <Text>Login</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#4169E1',
    justifyContent: 'center',
  },
  body: {
    padding:16,
  },
  title: {
    marginLeft: 12,
    color: '#FCBF49'
  },
  button: {
    backgroundColor: "#FCBF49",
    alignItems: "center",
    margin: 12,
    padding: 12,
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    borderColor: '#FCBF49',
    padding: 10,
    color: '#FCBF49',
  },
});

export default LoginScreen;