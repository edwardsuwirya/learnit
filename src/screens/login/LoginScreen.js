import React from 'react';
import { StyleSheet, SafeAreaView, View, Text, TextInput, TouchableOpacity } from 'react-native';


const LoginScreen = (props) => {
  const {
    userName,
    password,
    handleUserNameChange,
    handlePasswordChange,
    onAuthenticate
  } = props.bloc();

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.body}>
        <Text style={styles.title} >User Name</Text>
        <TextInput testID="userNameInput" style={styles.input}
          onChangeText={handleUserNameChange}
          value={userName} />
        <Text style={styles.title}>Password</Text>
        <TextInput testID="passwordInput" style={styles.input}
          secureTextEntry={true}
          onChangeText={handlePasswordChange}
          value={password} />
        <TouchableOpacity
          style={styles.button} onPress={() => onAuthenticate(userName, password, props.onNavigate)}>
          <Text>Login</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}



const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#4169E1',
    justifyContent: 'center',
  },
  body: {
    padding: 16,
  },
  title: {
    marginLeft: 12,
    color: '#FCBF49'
  },
  button: {
    backgroundColor: "#FCBF49",
    alignItems: "center",
    margin: 12,
    padding: 12,
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    borderColor: '#FCBF49',
    padding: 10,
    color: '#FCBF49',
  },
});

export default LoginScreen;