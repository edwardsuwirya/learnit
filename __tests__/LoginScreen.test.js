import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import LoginScreen from '../src/screens/login/LoginScreen';


describe('Login Screen', () => {
  const bloc = jest.fn();
  bloc.mockReturnValue({
    userName: '',
    password: '',
    handleUserNameChange: () => { },
    handlePasswordChange: () => { },
    onAuthenticate: () => {
      return { "token": "123" }
    }
  });

  it('renders correctly', () => {
    const { toJSON } = render(<LoginScreen bloc={bloc} />);
    expect(toJSON()).toMatchSnapshot()
  });

  //alternatif dari match snapshot
  it('renders form correctly', () => {
    const { getByTestId, queryByText } = render(<LoginScreen bloc={bloc} />);
    expect(queryByText('User Name')).not.toBeNull();
    expect(queryByText('Password')).not.toBeNull();
    expect(getByTestId('userNameInput')).not.toBeNull();
    expect(getByTestId('passwordInput')).not.toBeNull();
    expect(queryByText('Login')).not.toBeNull();
  });

  it('User do login', () => {
    const bloc = jest.fn();
    const onAuthenticateMock = jest.fn();
    bloc.mockReturnValue({
      userName: '',
      password: '',
      handleUserNameChange: jest.fn(),
      handlePasswordChange: jest.fn(),
      onAuthenticate: onAuthenticateMock
    });
    const { getByTestId, getByText } = render(<LoginScreen bloc={bloc} />);
    const userNameInput = getByTestId('userNameInput');
    fireEvent.changeText(userNameInput, 'doni');

    const passwordInput = getByTestId('passwordInput');
    fireEvent.changeText(passwordInput, '333');

    const loginButton = getByText('Login');
    fireEvent.press(loginButton);
    expect(onAuthenticateMock.mock.calls.length).toBe(1);
  });
});
