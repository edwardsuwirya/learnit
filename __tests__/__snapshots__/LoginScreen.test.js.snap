// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#4169E1",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FCBF49",
          "marginLeft": 12,
        }
      }
    >
      User Name
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#FCBF49",
          "borderWidth": 1,
          "color": "#FCBF49",
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
      testID="userNameInput"
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "#FCBF49",
          "marginLeft": 12,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "#FCBF49",
          "borderWidth": 1,
          "color": "#FCBF49",
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
      testID="passwordInput"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCBF49",
          "margin": 12,
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text>
        Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
